MbDotNet/IClient.cs

```diff
@@ -1,103 +1,103 @@
-        /// <exception cref="MbDotNet.Exceptions.InvalidProtocolException">Thrown if the retrieved imposter was not an HTTP imposter</exception>
-        RetrievedTcpImposter GetTcpImposter(int port);
-
-        /// <summary>
-        /// Retrieves an HttpsImposter along with information about requests made to that
-        /// imposter if mountebank is running with the "--mock" flag.
-        /// </summary>
-        /// <param name="port">The port number of the imposter to retrieve</param>
-        /// <returns>The retrieved imposter</returns>
-        /// <exception cref="MbDotNet.Exceptions.ImposterNotFoundException">Thrown if no imposter was found on the specified port.</exception>
-        /// <exception cref="MbDotNet.Exceptions.InvalidProtocolException">Thrown if the retrieved imposter was not an HTTP imposter</exception>
-        RetrievedHttpsImposter GetHttpsImposter(int port);
-
-        /// <summary>
-        /// Deletes a single imposter from mountebank. Will also remove the imposter from the collection
-        /// of imposters that the client maintains.
-        /// </summary>
-        /// <param name="port">The port number of the imposter to be removed</param>
-        void DeleteImposter(int port);
-
-        /// <summary>
-        /// Deletes all imposters from mountebank. Will also remove the imposters from the collection
-        /// of imposters that the client maintains.
-        /// </summary>
-        void DeleteAllImposters();
-
-        /// <summary>
-        /// Submits all pending imposters from the supplied collection to be created in mountebank. 
-        /// <exception cref="MbDotNet.Exceptions.MountebankException">Thrown if unable to create the imposter.</exception>
-        /// </summary>
-        void Submit(ICollection<Imposter> imposters);
-
-        /// <summary>
-        /// Submits imposter to be created in mountebank. 
-        /// <exception cref="MbDotNet.Exceptions.MountebankException">Thrown if unable to create the imposter.</exception>
-        /// </summary>
-        void Submit(Imposter imposter);
-    }
-}
+﻿using System.Collections.Generic;
+
+using MbDotNet.Enums;
+using MbDotNet.Models.Imposters;
+
+namespace MbDotNet
+{
+    public interface IClient
+    {
+        /// <summary>
+        /// A collection of all of the current imposters. The imposters in this
+        /// collection may or may not have been added to mountebank. See IImposter.PendingSubmission
+        /// for more information.
+        /// </summary>
+        ICollection<Imposter> Imposters { get; }
+
+        /// <summary>
+        /// Creates a new imposter on the specified port with the HTTP protocol. The Submit method
+        /// must be called on the client in order to submit the imposter to mountebank.
+        /// </summary>
+        /// <param name="port">The port the imposter will be set up to receive requests on</param>

```

Comment:
Could you please update the comments on the affected methods to signify what will happen if they do not provide a value for the port?